library/pool.c

```diff
@@ -1184,46 +1197,129 @@ int dnet_io_init(struct dnet_node *n, struct dnet_config *cfg)
 	return err;
 }
 
+int dnet_server_backend_init(struct dnet_node *n, size_t backend_id)
```

Comment:
Main reason of choosing array instead of tree - performance: backend lookup occurs every time in dnet_schedule_io().